packages/components/src/FilterBar/context/reducer/updateSingleFilter.ts

```diff
@@ -0,0 +1,22 @@
+import { FilterState, FiltersState, FiltersValues } from "../types"
+
+export const updateSingleFilter = <ValuesMap extends FiltersValues>(
+  state: FiltersState<ValuesMap>,
+  id: keyof ValuesMap,
+  data: Partial<FilterState<keyof ValuesMap, ValuesMap[keyof ValuesMap]>>
+): FiltersState<ValuesMap> => {
+  const newFilterState = { ...state.filters[id], ...data }
+
+  if (data.isActive !== undefined) {
+    if (data.isActive) {
+      state.activeFilterIds.add(id)
```

Comment:
Would it be easier to destructure state?

Reply:
I can try. Wasn't sure if it would throw an error if it was undefined (although now that I think about it, it should only care if `data` didn't exist, not `isActive`).